storage/storage-interface/src/lib.rs

```diff
@@ -642,6 +668,33 @@ pub trait DbWriter: Send + Sync {
     }
 }
 
+pub fn get_state_value_resolver_for_version(
+    db_reader: Arc<dyn DbReader>,
+    version: Version,
+) -> Box<dyn Fn(&StateKey) -> anyhow::Result<Option<Vec<u8>>>> {
+    Box::new(move |x| {
+        let state_value_with_proof = db_reader.get_state_value_with_proof_by_version(x, version)?;
+        if let Some(state_value) = state_value_with_proof.0 {
+            return Ok(state_value.maybe_bytes.as_ref().cloned());
+        } else {
+            Ok(None)
+        }
+    })
+}
+
+pub fn get_state_value_resolver_for_latest_version(
+    db_reader: Arc<dyn DbReader>,
+) -> Box<dyn Fn(&StateKey) -> anyhow::Result<Option<Vec<u8>>>> {
```

Comment:
why we need a closure for this? can't we use static function?

Reply:
@zekun000 - `AccountStateView` needs to accept a closure to cater to various needs.  `AccountStateView` is currently used in many different scenarios as follows

1. StateCache (HashMap) to provide this view
2. DB reader to provide latest state view.
3. DB reader to provide state view for a version.
4. [TBD] AccountState to provide this view to remove redundant functions in AccountState.

Let me know if you have any better idea to structure this.